plugins/gcp/fix_plugin_gcp/collector.py

```diff
@@ -131,6 +159,18 @@ def collect(self) -> None:
             self.core_feedback.progress_done(self.project.id, 1, 1, context=[self.cloud.id])
             log.info(f"[GCP:{self.project.id}] Collecting resources done.")
 
+    def collect_usage_metrics(self, builder: GraphBuilder) -> None:
+        for resource in builder.graph.nodes:
+            if isinstance(resource, GcpResource) and (mq := resource.collect_usage_metrics(builder)):
+                start_at = builder.created_at - builder.metrics_delta
+                region = cast(GcpRegion, resource.region())
+                try:
+                    rb = builder.for_region(region)
+                    result = monitoring.GcpMonitoringMetricData.query_for(rb, mq, start_at, builder.created_at)
+                    monitoring.update_resource_metrics(resource, result)
```

Comment:
No. This way, you fetch metrics one by one.
You should call this method, pass the resource and not wait for the result.
The called method can then spawn threads etc.